### PR TITLE
readme, docs: refine language for 5 documents

### PR DESCRIPTION
**Pull request by TomShawn**
### What problem does this PR solve? 

This PR improves the language and format of 5 TiSpark documents for better readability. 

### What is changed and how it works?

N/A

### CODE REVIEW

File: docs/datasource_api_userguide.md [hunk 1]
```diff
@@ -343,6 +370,7 @@ The full conversion metrics is as follows.
 | SET        | :x:                | :x:                | :x:                | :x:                | :x:                | :x:                | :x:                | :x:                | :x:                | :x:                | :x:                |
 
 ## Write Benchmark
+
 Tested on 4 machines as follows:
```
Comment (TomShawn) [inline docs/datasource_api_userguide.md]: What has been tested on 4 machines? What is the subject of the sentence?

File: docs/datasource_api_userguide.md [hunk 2]
```diff
@@ -1,147 +1,161 @@
...
-  .select("C_NAME")
-df.show()
-```
+Because TiDB is a database that supports `transaction`, The TiDB Connector for Spark also supports `transaction`, which means that:
 
-## Using the Spark Connector Without Extensions Enabled
-Let's see how to use the connector without extensions enabled.
+1. all the data in DataFrame is written to TiDB successfully if no conflicts exist.
+2. no data in DataFrame is written to TiDB successfully if conflicts exist.
+3. no partial changes are visible to other sessions until commit.
```
Comment (TomShawn) [inline docs/datasource_api_userguide.md]: What does it mean by "until commit"?

File: docs/datasource_api_userguide.md (hunk 1 above)
Comment (TomShawn) [inline docs/datasource_api_userguide.md]: What has been tested on 4 machines? What is the subject of the sentence?

File: docs/datasource_api_userguide.md [hunk 3]
```diff
@@ -242,73 +260,82 @@ CREATE TABLE tpch_test.TARGET_TABLE_CUSTOMER (
...
-| enableRegionSplit           | -             | false    | do region split to avoid hot region during insertion                                                        | true    |
-| regionSplitNum              | -             | false    | user defined region split number during insertion                                                           | 0       |
-| replace                     | -             | false    | define the behavior of append.                                                                              | false   |
-| lockTTLSeconds              | -             | false    | tikv lock ttl, write duration must < `lockTTLSeconds`, otherwise write may fail because of gc               | 3600    |
-| writeConcurrency            | -             | false    | maximum number of threads writing data to tikv, suggest `writeConcurrency` <= 8 * `number of TiKV instance` | 0       |
+| skipCommitSecondaryKey      | -             | false    | To skip commit secondary key                                                                                   | false   |
```
Comment (TomShawn) [inline docs/datasource_api_userguide.md]: What does it mean by "skip commit secondary key"? looks like it makes no sense grammatically.

File: docs/datasource_api_userguide.md [hunk 4]
```diff
@@ -242,73 +260,82 @@ CREATE TABLE tpch_test.TARGET_TABLE_CUSTOMER (
...
 
-# When create table, split a separated region for it. It is recommended to
-# turn off this option if there will be a large number of tables created.
+# When creating table, split a separated Region for it. It is recommended to
+# turn off this option if there is a large number of tables created.
 split-table: true
 ```
 
-If your TiDB's version is < 3.0 and want to have a try, you can set `spark.tispark.write.without_lock_table` to `true` to enable write, but no ACID is guaranteed.
+If your TiDB's version is earlier than 3.0 and you still want to have a try, set `spark.tispark.write.without_lock_table` to `true` to enable write, but ACID is **not** guaranteed.
```
Comment (TomShawn) [inline docs/datasource_api_userguide.md]: "you still want to have a try"... Try what? The meaning here is not so clear.

File: README.md [hunk 5]
```diff
@@ -130,80 +146,96 @@ ti.tidbMapDatabase ("tpch_test")
...
-| spark.tispark.show_rowid |  false | If to show implicit row Id if exists |
-| spark.tispark.db_prefix |  "" | A string indicating the extra database prefix for all databases in TiDB to distinguish them from Hive databases with the same name |
-| spark.tispark.request.isolation.level |  "SI" | Isolation level means whether do the resolve lock for the underlying tidb clusters. When you use the "RC", you will get the newest version of record smaller than your tso and ignore the locks. And if you use "SI", you will resolve the locks and get the records according to whether the resolved lock is committed or aborted  |
-
-## Log4j Configuration
+| `spark.tispark.pd.addresses` |  `127.0.0.1:2379` | The addresses of PD cluster, which are split by comma |
+| `spark.tispark.grpc.framesize` |  `268435456` | The maximum frame size of gRPC response |
+| `spark.tispark.grpc.timeout_in_sec` |  `10` | The gRPC timeout time in seconds |
+| `spark.tispark.meta.reload_period_in_sec` |  `60` | The reloading period of metastore in seconds |
+| `spark.tispark.plan.allow_agg_pushdown` |  `true` | If allow aggregation pushdown (in case of busy TiKV nodes) |
```
Comment (TomShawn) [inline README.md]: What does it mean by "If allow aggregation pushdown"?

File: README.md [hunk 6]
```diff
@@ -130,80 +146,96 @@ ti.tidbMapDatabase ("tpch_test")
...
+| `spark.tispark.plan.allow_index_read` |  `true` |  If allow index read (which might cause heavy pressure on TiKV) |
```
Comment (TomShawn) [inline README.md]: What does it mean by "If allow index read"?

File: docs/userguide.md [hunk 7]
```diff
@@ -1,205 +1,206 @@
...
 
-+ The current version of TiSpark supports Spark 2.3+/2.4+. It does not support any versions earlier than Spark 2.3.
-+ TiSpark requires JDK 1.8+ and Scala 2.11 (Spark2.0 + default Scala version).
-+ TiSpark runs in any Spark mode such as YARN, Mesos, and Standalone.
+TiSpark relies on the availability of TiKV clusters and PDs. You also need to set up and use the Spark clustering platform.
 
+## Prerequisites for setting up TiSpark
 
-## Recommended configuration
++ The current TiSpark version supports Spark 2.3+/2.4+, but does not support any Spark versions earlier than 2.3.
```
Comment (TomShawn) [inline docs/userguide.md]: What is the current TiSpark version?

File: README.md [hunk 8]
```diff
@@ -34,22 +34,33 @@ If you are using maven(recommended), add the following to your pom.xml:
 </dependencies>
 ```
 
-For other build tools, you can visit search.maven.org and search with GroupId [![Maven Search](https://img.shields.io/badge/com.pingcap-tikv/tispark-green.svg)](http://search.maven.org/#search%7Cga%7C1%7Cpingcap)(This search will also list all available modules of TiSpark including tikv-client).
+For other build tools, visit <https://search.maven.org/> and search with GroupId [![Maven Search](https://img.shields.io/badge/com.pingcap-tikv/tispark-green.svg)](http://search.maven.org/#search%7Cga%7C1%7Cpingcap). This search also lists all the available modules of TiSpark including tikv-client.
 
 ## How to build from sources
-TiSpark now supports Spark 2.3.0+/2.4.0+. The previous version for Spark 2.1.0+ will only contain bug fixes in future, you may still get Spark 2.1 support until TiSpark 1.2.1.
+
+TiSpark now supports Spark 2.3.0+ or 2.4.0+. The earlier TiSpark versions for Spark 2.1.0+ only contains bug-fixes. After these versions, you can still get support for Spark 2.1 until TiSpark 1.2.1.
```
Comment (birdstorm) [inline README.md]: contains -> contain? or versions -> version

File: README.md [hunk 9]
```diff
@@ -60,20 +71,20 @@ Remember to add `-Dmaven.test.skip=true` to skip all the tests if you don't need
 | Spark-2.2.x | TiSpark-1.2.1 |
 | Spark-2.1.x | TiSpark-1.2.1 |
 
-## Maximum TiDB/TiKV/PD version supported by TiSpark
+## Latest TiDB/TiKV/PD versions supported by TiSpark
 
-Each latest TiSpark version guarantees *backward compatibility* for TiDB components, i.e., supports TiDB/TiKV/PD until a certain release. Its reason varies, amongst which the most common one is that the new features and bug-fixes provided by TiDB components require an update on API, dependencies, etc.
+Each latest TiSpark version guarantees *backward compatibility* to TiDB components. That is to say that TiSpark supports TiDB/TiKV/PD versions all the way back to an earlier release of these components. The reason varies, amongst which the most common one is that the new features and bug-fixes provided by TiDB components require an update on API, dependencies, etc.
```
Comment (birdstorm) [inline README.md]: > That is to say that TiSpark supports TiDB/TiKV/PD versions all the way back to an earlier release of these components.

Is this statement true? @marsishandsome

File: README.md (hunk 8 above)
Comment (TomShawn) [reply README.md]: typo fixed.

File: README.md (hunk 5 above)
Comment (birdstorm) [reply README.md]: should be whether `aggregations are allowed to push down to TiKV`

File: README.md (hunk 6 above)
Comment (birdstorm) [reply README.md]: should be `whether index is enabled in planning`

File: README.md [hunk 10]
```diff
@@ -130,80 +146,96 @@ ti.tidbMapDatabase ("tpch_test")
...
-## Compatibility with tidb-3.0
+For how TiSpark can benefit from TiDB's statistic information, see [here](./docs/userguide.md).
+
+## Compatibility with TiDB 3.0
+
 ### View
-TiDB starts to support `view` from `tidb-3.0`.
 
-TiSpark currently **does not support** `view`. Users will not be able to observe or access data through views with TiSpark.
+TiDB starts to support `view` from `tidb-3.0` on.
```
Comment (birdstorm) [inline README.md]: how about we change it to `since`

File: README.md [hunk 11]
```diff
@@ -130,80 +146,96 @@ ti.tidbMapDatabase ("tpch_test")
...
 
 ### Table Partition
+
 `tidb-3.0` supports both `Range Partition` and `Hash Partition`.
 
-TiSpark currently **supports** `Range Partition` and `Hash Partition`. Users can select data from `Range Partition` table and `Hash Partition` table through TiSpark.
+TiSpark currently supports `Range Partition` and `Hash Partition`. Users can select data from the `Range Partition` table and the `Hash Partition` table through TiSpark.
 
-In most cases, TiSpark will use a full table scan. Only in some cases, TiSpark will apply partition pruning (read more [here](./docs/userguide.md).
+In most cases, TiSpark use a full table scan. Only in certain cases, TiSpark applies partition pruning. For more details, see [here](./docs/userguide.md).
```
Comment (birdstorm) [inline README.md]:
```suggestion
In most cases, TiSpark use a full table scan on partition tables. Only in certain cases, TiSpark applies partition pruning. For more details, see [here](./docs/userguide.md).
```

File: README.md (hunk 10 above)
Comment (TomShawn) [reply README.md]: No problem.

File: docs/datasource_api_userguide.md (hunk 2 above)
Comment (birdstorm) [reply docs/datasource_api_userguide.md]: `until committed`? It does seems awkward here. I will discuss with @marsishandsome about this chapter.

File: docs/datasource_api_userguide.md (hunk 3 above)
Comment (birdstorm) [reply docs/datasource_api_userguide.md]: whether to skip the commit phase of secondary keys.

File: docs/datasource_api_userguide.md [hunk 12]
```diff
@@ -242,73 +260,82 @@ CREATE TABLE tpch_test.TARGET_TABLE_CUSTOMER (
...
-| writeConcurrency            | -             | false    | maximum number of threads writing data to tikv, suggest `writeConcurrency` <= 8 * `number of TiKV instance` | 0       |
+| skipCommitSecondaryKey      | -             | false    | To skip commit secondary key                                                                                   | false   |
+| enableRegionSplit           | -             | false    | To split Region to avoid hot Region during insertion                                                        | true    |
+| regionSplitNum              | -             | false    | The Region split number defined by user during insertion                                                           | 0       |
+| replace                     | -             | false    | To define the behavior of append                                                                              | false   |
+| lockTTLSeconds              | -             | false    | tikv lock ttl. The write duration must be longer than `lockTTLSeconds`, otherwise write might fail because of the Garbage Collection (GC).              | 3600    |
```
Comment (birdstorm) [inline docs/datasource_api_userguide.md]:
```suggestion
| lockTTLSeconds              | -             | false    | TiKV's lock TTL. The write duration must be no longer than `lockTTLSeconds`, otherwise write might fail because of the Garbage Collection (GC).              | 3600    |
```

File: docs/datasource_api_userguide.md (hunk 4 above)
Comment (birdstorm) [reply docs/datasource_api_userguide.md]: Just delete `and you still want to have a try`...

File: docs/datasource_api_userguide.md [hunk 13]
```diff
@@ -242,73 +260,82 @@ CREATE TABLE tpch_test.TARGET_TABLE_CUSTOMER (
...
+The following types of SparkSQL Data are currently not supported for writing to TiDB:
+
 - BinaryType
 - ArrayType
 - MapType
 - StructType
 
-The full conversion metrics is as follows.
-
+The complete conversion metrics is as follows.
```
Comment (birdstorm) [inline docs/datasource_api_userguide.md]: I think metrics should be plural when used as "measurements"

File: docs/datasource_api_userguide.md (hunk 1 above)
Comment (birdstorm) [reply docs/datasource_api_userguide.md]: @marsishandsome  PTAL

File: docs/userguide.md [hunk 14]
```diff
@@ -1,205 +1,206 @@
...
+> **Note:**
+>
+> This is a user guide for TiSpark version >= 2.0. If you are using version earlier than 2.0, refer to [Document for Spark 2.1](./userguide_spark2.1.md)
 
+This document introduces how to set up and use TiSpark, which requires some basic knowledge of Apache Spark. Refer to [Spark website](https://spark.apache.org/docs/latest/index.html) for details.
 
 ## Overview
 
-TiSpark is an OLAP solution that runs Spark SQL directly on TiKV, the distributed storage engine.
+TiSpark is a thin layer built for running Apache Spark on top of TiDB/TiKV to answer the complex OLAP queries. It enjoys the merits of both the Spark platform and the distributed clusters of TiKV, at the same time, seamlessly integrated to TiDB, the distributed OLTP database, to provide one-stop Hybrid Transactional/Analytical Processing (HTAP) solutions for online transactions and analyses.
```
Comment (birdstorm) [inline docs/userguide.md]: Such a long composite sentence...

Some suggestions:
- `integrated -> integrates`? If you mean by `it enjoys the merits while integrating to tidb`.
- `to provide -> provides`? If you mean by `it enjoys the merits while providing solutions`.
- Or change `at the same time` to `via/by`? If the following is considered an attributive clause.
- Or we may simply split this sentence into two for better understanding...

File: docs/userguide.md (hunk 7 above)
Comment (birdstorm) [reply docs/userguide.md]: current version of TiSpark.

File: docs/userguide.md [hunk 15]
```diff
@@ -1,205 +1,206 @@
...
 
-Similarly, you can start a Spark-Slave node with the following command:
+Similarly, start a Spark-Slave node by executing the following command:
 
 ```
 ./sbin/start-slave.sh spark://spark-master-hostname:7077
 ```
 
-After the command returns, you can see if the Slave node is joined to the Spark cluster correctly from the panel as well. Repeat the above command at all Slave nodes. After all the slaves are connected to the master, you have a Standalone mode Spark cluster.
+After the command returns, also see whether the Slave node is joined to the Spark cluster correctly from the panel.
```
Comment (birdstorm) [inline docs/userguide.md]: `see -> check`?

File: README.md (hunk 9 above)
Comment (zhexuany) [reply README.md]: I doubt it...

File: docs/userguide.md (hunk 15 above)
Comment (TomShawn) [reply docs/userguide.md]: `check` is better.

File: docs/userguide.md [hunk 16]
```diff
@@ -236,106 +239,128 @@ select count(*) from account;
...
+Q: What are the pros and cons of independent deployment as opposed to a shared resource with an existing Spark / Hadoop cluster?
 
 A: You can use the existing Spark cluster without a separate deployment, but if the existing cluster is busy, TiSpark will not be able to achieve the desired speed.
 
 Q: Can I mix Spark with TiKV?
 
-A: If TiDB and TiKV are overloaded and run critical online tasks, consider deploying TiSpark separately. You also need to consider using different NICs to ensure that OLTP's network resources are not compromised and affect online business. If the online business requirements are not high or the loading is not large enough, you can consider mixing TiSpark with TiKV deployment.
+A: If TiDB and TiKV are overloaded and run critical online tasks, consider deploying TiSpark separately.
+
+You also need to consider using different NICs to ensure that OLTP's network resources are not compromised and do not affect online business.
```
Comment (birdstorm) [inline docs/userguide.md]:
```suggestion
You also need to consider using different NICs to ensure that OLTP's network resources are not compromised so that online business is not affected.
```

File: docs/datasource_api_userguide.md [hunk 17]
```diff
@@ -242,73 +260,82 @@ CREATE TABLE tpch_test.TARGET_TABLE_CUSTOMER (
...
 
-select data from `tpch_test.CUSTOMER`:
+4. Select data from `tpch_test.CUSTOMER`:
+
 ```
 SELECT * FROM CUSTOMER_SRC limit 10
 ```
 
-register another tidb table `tpch_test.TARGET_TABLE_CUSTOMER` to spark catalog:
+5. Register another TiDB table `tpch_test.TARGET_TABLE_CUSTOMER` to the spark catalog:
```
Comment (zhexuany) [inline docs/datasource_api_userguide.md]: s/spark catalog/Spark Catalog

File: docs/userguide.md (hunk 14 above)
Comment (TomShawn) [reply docs/userguide.md]: What about `While enjoying the merits of both the Spark platform and the distributed clusters of TiKV, it is seamlessly integrated into TiDB, the distributed OLTP database, and provide one-stop Hybrid Transactional/Analytical Processing (HTAP) solutions for online transactions and analyses.`?

File: docs/datasource_api_userguide.md [hunk 18]
```diff
@@ -242,73 +260,82 @@ CREATE TABLE tpch_test.TARGET_TABLE_CUSTOMER (
 )
 ```
 
-register a tidb table `tpch_test.CUSTOMER` to spark catalog:
+3. Register a TiDB table `tpch_test.CUSTOMER` to the Spark catalog:
```
Comment (zhexuany) [inline docs/datasource_api_userguide.md]: s/spark catalog/Spark Catalog

File: docs/datasource_api_userguide.md [hunk 19]
```diff
@@ -194,8 +210,8 @@ df.show()
 */
 
 // Common options can also be passed in,
-// e.g. spark.tispark.plan.allow_agg_pushdown, spark.tispark.plan.allow_index_read, etc.
-// spark.tispark.plan.allow_index_read is optional
+// for example, spark.tispark.plan.allow_agg_pushdown, spark.tispark.plan.allow_index_read, etc.
```
Comment (zhexuany) [inline docs/datasource_api_userguide.md]: Is this change necessary here?

File: docs/userguide.md (hunk 14 above)
Comment (birdstorm) [reply docs/userguide.md]: Great! Just replace `it` with `TiSpark`.

File: docs/datasource_api_userguide.md [hunk 20]
```diff
@@ -1,147 +1,161 @@
...
-  .format("tidb")
-  .options(tidbOptions)
-  .option("database", "tpch_test")
-  .option("table", "CUSTOMER")
-  .load()
-  .filter("C_CUSTKEY = 1")
-  .select("C_NAME")
-df.show()
-```
+Because TiDB is a database that supports `transaction`, The TiDB Connector for Spark also supports `transaction`, which means that:
```
Comment (zhexuany) [inline docs/datasource_api_userguide.md]: This pretty weird. @birdstorm We need to discuss the entire paragraph.

Comment (birdstorm) [reply docs/datasource_api_userguide.md]: That's what I think. We will discuss the entire chapter before reaching an agreement.

File: docs/userguide.md (hunk 14 above)
Comment (birdstorm) [reply docs/userguide.md]: I'm just wondering that `providing one-stop solutions` may be a result of `integrated into TiDB`, so that they should not conjunct as parallel.

File: docs/datasource_api_userguide.md (hunk 19 above)
Comment (TomShawn) [reply docs/datasource_api_userguide.md]: We avoid using these abbreviations in our official document such as `i.e.`, `e.g.`, `etc.`.

File: docs/userguide.md (hunk 14 above)
Comment (birdstorm) [reply docs/userguide.md]: Also, I believe `integrate with` is better than `integrate into`. @TomShawn

Comment (TomShawn) [reply docs/userguide.md]: Yes, you are right. What about `While enjoying the merits of both the Spark platform and the distributed clusters of TiKV, it is seamlessly integrated with TiDB, the distributed OLTP database, and thus blessed to provide one-stop Hybrid Transactional/Analytical Processing (HTAP) solutions for online transactions and analyses.`?

Comment (birdstorm) [reply docs/userguide.md]: nice

File: docs/datasource_api_userguide.md (hunk 2 above)
Comment (marsishandsome) [reply docs/datasource_api_userguide.md]: @TomShawn 在commit之前其他session不会看到`修改的数据`

File: docs/datasource_api_userguide.md (hunk 1 above)
Comment (marsishandsome) [reply docs/datasource_api_userguide.md]: @TomShawn 这个测试报告是在4台机器上进行测试的

File: README.md (hunk 9 above)
Comment (marsishandsome) [reply README.md]: 建议可以把这段删了，下面的表格已经表达了这个意思，并且更精确

File: docs/datasource_api_userguide.md (hunk 20 above)
Comment (marsishandsome) [reply docs/datasource_api_userguide.md]: i will fix in next PR.
